tests/type-destructors/__snapshots__/jsfmt.spec.js.snap

```diff
@@ -69,6 +69,7 @@ var x1: $NonMaybeType<number | string> = true;
 // err, boolean ~> number|string
 var x2: $PropertyType<{ p: number } | { p: string }, \"p\"> = 0;
 // ok, number ~> number|string
-var x3: $PropertyType<{ p: number } |
-  { p: string }, \"p\"> = true; // err, boolean ~> number|string"
+var x3: $PropertyType<
```

Comment:
I'd say it's fine for now. The common case this is solving is worth it and we can figure out these special cases later.